Compile error using MinGW-GCC in win10

I try to install lusocket via luarocks. But it fail compiling. MinGW-GCC define `socklen_t` as `size_t`. Is MinGW compiler supported?
```
$ luarocks install luasocket
In file included from src/luasocket.c:30:
src/inet.h:46:13: error: conflicting types for 'inet_ntop'
   46 | const char *inet_ntop(int af, const void *src, char *dst, socklen_t cnt);
      |             ^~~~~~~~~
In file included from src/wsocket.h:12,
                 from src/socket.h:18,
                 from src/inet.h:18,
                 from src/luasocket.c:30:
C:/msys64/mingw64/x86_64-w64-mingw32/include/ws2tcpip.h:451:35: note: previous declaration of 'inet_ntop' was here
  451 | WINSOCK_API_LINKAGE LPCSTR WSAAPI InetNtopA(INT Family, LPCVOID pAddr, LPSTR pStringBuf, size_t StringBufSize);
      |                                   ^~~~~~~~~
In file included from src/luasocket.c:30:
src/inet.h:47:5: warning: 'inet_pton' redeclared without dllimport attribute: previous dllimport ignored [-Wattributes]
   47 | int inet_pton(int af, const char *src, void *dst);
      |     ^~~~~~~~~
Installing https://luarocks.org/luasocket-3.0rc1-2.src.rock
gcc -O2 -fPIC -c -o src/mime.o -IC:/msys64/mingw64/include/lua5.1 src/mime.c -DLUA_COMPAT_APIINTCASTS -DLUASOCKET_DEBUG -DLUASOCKET_INET_PTON -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport)
gcc -shared -o mime/core.dll src/mime.o C:/msys64/mingw64/lib/liblua5.1.dll.a -lm
gcc -O2 -fPIC -c -o src/luasocket.o -IC:/msys64/mingw64/include/lua5.1 src/luasocket.c -DLUA_COMPAT_APIINTCASTS -DLUASOCKET_DEBUG -DLUASOCKET_INET_PTON -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport)

Error: Build error: Failed compiling object src/luasocket.o
```

The issue is still there. It can be fixed by removing option `LUASOCKET_INET_PTON` compilation directive.

However, I run into the following errors : 
```
$ luarocks.lua make
src/luasocket.c: In function 'global_skip':
src/luasocket.c:67:18: warning: implicit declaration of function 'luaL_checkint'; did you mean 'luaL_checkany'? [-Wimplicit-function-declaration]
   67 |     int amount = luaL_checkint(L, 1);
      |                  ^~~~~~~~~~~~~
      |                  luaL_checkany
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/lib/../lib/dllcrt2.o: in function `_CRT_INIT':
C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:130: undefined reference to `_execute_onexit_table'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/lib/../lib/dllcrt2.o: in function `pre_c_init':
C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:64: undefined reference to `_initialize_onexit_table'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/lib/../lib/dllcrt2.o: in function `atexit':
C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:206: undefined reference to `_register_onexit_function'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: src/luasocket.o:luasocket.c:(.text+0x20): undefined reference to `luaL_checkint'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: src/inet.o:inet.c:(.text+0x7ec): undefined reference to `gai_strerrorA'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: src/inet.o:inet.c:(.text+0x94c): undefined reference to `gai_strerrorA'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: src/udp.o:udp.c:(.text+0x7dc): undefined reference to `gai_strerrorA'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: src/udp.o:udp.c:(.text+0xab4): undefined reference to `gai_strerrorA'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: src/wsocket.o:wsocket.c:(.text+0xe81): undefined reference to `gai_strerrorA'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/lib/../lib/libmingw32.a(lib64_libmingw32_a-pseudo-reloc.o): in function `__report_error':
C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/pseudo-reloc.c:149: undefined reference to `__acrt_iob_func'
D:/utils/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/pseudo-reloc.c:150: undefined reference to `__acrt_iob_func'
collect2.exe: error: ld returned 1 exit status

luasocket scm-0 depends on lua >= 5.1 (5.4-1 provided by VM)
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/luasocket.o -ID:/utils/msys64/mingw64/include src/luasocket.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/timeout.o -ID:/utils/msys64/mingw64/include src/timeout.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/buffer.o -ID:/utils/msys64/mingw64/include src/buffer.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/io.o -ID:/utils/msys64/mingw64/include src/io.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/auxiliar.o -ID:/utils/msys64/mingw64/include src/auxiliar.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/options.o -ID:/utils/msys64/mingw64/include src/options.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/inet.o -ID:/utils/msys64/mingw64/include src/inet.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/except.o -ID:/utils/msys64/mingw64/include src/except.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/select.o -ID:/utils/msys64/mingw64/include src/select.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/tcp.o -ID:/utils/msys64/mingw64/include src/tcp.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/udp.o -ID:/utils/msys64/mingw64/include src/udp.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -O2 -c -o src/wsocket.o -ID:/utils/msys64/mingw64/include src/wsocket.c -DLUASOCKET_DEBUG -DWINVER=0x0501 -DLUASOCKET_API=__declspec(dllexport) -DMIME_API=__declspec(dllexport) -Ic:/windows/system32/include
D:\edebry\Documents\utils\msys64\mingw64\bin\gcc.exe -shared -o socket/core.dll src/luasocket.o src/timeout.o src/buffer.o src/io.o src/auxiliar.o src/options.o src/inet.o src/except.o src/select.o src/tcp.o src/udp.o src/wsocket.o -Lc:/windows/system32 -lws2_32 D:/utils/msys64/mingw64/bin/lua54.dll -lm

Error: Build error: Failed compiling module socket/core.dll
```

--- a/rockspecs/luasocket-3.0.0-1.rockspec
+++ b/rockspecs/luasocket-3.0.0-1.rockspec
@@ -34,7 +34,7 @@ local function make_plat(plat)
     },
     mingw32 = {
       "LUASOCKET_DEBUG",
-      "LUASOCKET_INET_PTON",
+      -- "LUASOCKET_INET_PTON",
       "WINVER=0x0501"
     }
   }
@@ -113,6 +113,7 @@ local function make_plat(plat)
   then
     modules["socket.core"].sources[#modules["socket.core"].sources+1] = "src/wsocket.c"
     modules["socket.core"].libraries = { "ws2_32" }
+    modules["socket.core"].libdirs = {}
   end
   return { modules = modules }
 end

